source/extensions/filters/network/rbac/rbac_filter.cc

```diff
@@ -0,0 +1,64 @@
+#include "extensions/filters/network/rbac/rbac_filter.h"
+
+#include "envoy/buffer/buffer.h"
+#include "envoy/network/connection.h"
+
+#include "common/http/header_map_impl.h"
+
+#include "extensions/filters/network/well_known_names.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace RBACFilter {
+
+Network::FilterStatus RoleBasedAccessControlFilter::onData(Buffer::Instance&, bool) {
+  ENVOY_LOG(
+      debug, "checking connection: remoteAddress: {}, localAddress: {}, ssl: {}",
+      callbacks_->connection().remoteAddress()->asString(),
+      callbacks_->connection().localAddress()->asString(),
+      callbacks_->connection().ssl()
+          ? "uriSanPeerCertificate: " + callbacks_->connection().ssl()->uriSanPeerCertificate() +
+                ", subjectPeerCertificate: " +
+                callbacks_->connection().ssl()->subjectPeerCertificate()
+          : "none");
+
+  const Envoy::Http::HeaderMapImpl empty_header;
```

Comment:
onData is called multiple times.   we should only do rbac check on the first one.

Reply:
I can cache the check result, but just curious could it have any problems with the cache? I assume the peer certification, ip and port won't change on the following onData() call right? Also whenever a newer RBAC config is pushed, envoy will create a new network filter, so it's also no problem to cache the result?

Reply:
Envoy will call onData() for each buffer it receives.  But you don't need to call rbac check on every buffer.

Reply:
Done.